pkg/costattribution/tracker.go

```diff
@@ -0,0 +1,271 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package costattribution
+
+import (
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/model/labels"
+	"go.uber.org/atomic"
+)
+
+type Tracker interface {
```

Comment:
I think we don't need an interface here, it just makes things more complex and slow. Now we can't compare two trackers by just comparing the pointers.

There are only two implementations of Tracker: the one that does things and the one that is disabled, we can make the disabled one be the nil pointer of `TrackerImpl` (please rename that back to Tracker).

Reply:
removed the interface, and it checks for nil pointer in the receiver instead. commit 7f0b3722f0f5b0f51397f74505f370f6a12a2ffc